playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

```diff
@@ -150,16 +155,22 @@
 
         playerSettings = settings != null ? settings : new PlayerSettings();
         rootView = rootPlayerView;
-        DefaultBandwidthMeter.Builder bandwidthMeterBuilder = new DefaultBandwidthMeter.Builder(context);
 
-        Long initialBitrateEstimate = playerSettings.getAbrSettings().getInitialBitrateEstimate();
+        LoadControlStrategy customLoadControlStrategy = (LoadControlStrategy) playerSettings.getCustomLoadControlStrategy();
+        if (customLoadControlStrategy != null && customLoadControlStrategy.getCustomBandwidthMeter() != null) {
+            bandwidthMeter = customLoadControlStrategy.getCustomBandwidthMeter();
+        } else {
+            DefaultBandwidthMeter.Builder bandwidthMeterBuilder = new DefaultBandwidthMeter.Builder(context);
+
+            Long initialBitrateEstimate = playerSettings.getAbrSettings().getInitialBitrateEstimate();
 
-        if (initialBitrateEstimate != null && initialBitrateEstimate > 0) {
-            bandwidthMeterBuilder.setInitialBitrateEstimate(initialBitrateEstimate);
-        }
+            if (initialBitrateEstimate != null && initialBitrateEstimate > 0) {
+                bandwidthMeterBuilder.setInitialBitrateEstimate(initialBitrateEstimate);
+            }
 
-        bandwidthMeter = bandwidthMeterBuilder.build();
-        bandwidthMeter.addEventListener(mainHandler, this);
+            bandwidthMeter = bandwidthMeterBuilder.build();
+            bandwidthMeter.addEventListener(mainHandler, this);
+        }
```

Comment:
@noamtamim 

what if listener is not added does it have influence on profiler or maybe events we consume for bitrate events foe kava na youbora?

Reply:
Not just for the profiler -- this listener is required because we use it to report the PLAYBACK_INFO_UPDATED event.

@tri170391 please make sure this line is always executed, even for your `BandwidthMeter`. In addition, make sure to actually call this listener from the custom BandwidthMeter whenever the estimation changes.